src/transport/SessionManager.cpp

```diff
@@ -736,27 +748,21 @@ void SessionManager::SecureUnicastMessageDispatch(const PacketHeader & packetHea
     }
 }
 
-void SessionManager::SecureGroupMessageDispatch(const PacketHeader & packetHeader, const Transport::PeerAddress & peerAddress,
+void SessionManager::SecureGroupMessageDispatch(PacketHeader & packetHeader, const Transport::PeerAddress & peerAddress,
                                                 System::PacketBufferHandle && msg)
 {
     PayloadHeader payloadHeader;
+    PacketHeader packetHeaderCopy;
+    System::PacketBufferHandle msgCopy;
     Credentials::GroupDataProvider * groups = Credentials::GetGroupDataProvider();
     VerifyOrReturn(nullptr != groups);
     CHIP_ERROR err = CHIP_NO_ERROR;
 
-    if (!packetHeader.GetDestinationGroupId().HasValue())
+    if (!packetHeader.HasDestinationGroupId())
     {
         return; // malformed packet
     }
 
-    GroupId groupId = packetHeader.GetDestinationGroupId().Value();
-
-    if (msg.IsNull())
```

Comment:
Is this check no longer needed and if so, why?

Reply:
It moved to line 809 below.

Reply:
Ah, it got moved down. Nonetheless, why wouldn't we do the check here? Doesn't that ensure that we're not accessing out of bounds memory later below?

Reply:
No, the check is invalid until `DecodeAndConsume` is called.  Also, the `DecodeFixed` does checks that the minimum sized header exists.  I suppose we could possibly move the check into DecodeFixed and DecodeAndConsume (that create a new assumption that zero payload messages are invalid globally), but the philosophy for this PR was to minimize changes to the existing message processing.

Reply:
Added IsNull check after msgClone.